dmscripts/notify_suppliers_of_new_questions_answers.py

```diff
@@ -0,0 +1,159 @@
+    responses = data_api_client.find_brief_responses(brief_id=brief["id"])
+    return [response["supplierId"] for response in responses["briefResponses"]]
+
+
+def get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief):
+    audit_events = data_api_client.find_audit_events(
+        audit_type=dmapiclient.audit.AuditTypes.send_clarification_question,
+        object_type='briefs',
+        object_id=brief['id']
+    )
+    return [audit_event['data']['supplierId'] for audit_event in audit_events['auditEvents']]
+
+
+def get_ids_of_interested_suppliers_for_briefs(data_api_client, briefs):
+    interested_suppliers = {}
+    for brief in briefs:
+        suppliers_who_applied = get_ids_of_suppliers_who_started_applying(data_api_client, brief)
+        suppliers_who_asked_a_question = get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief)
+        interested_suppliers[brief['id']] = list(set(suppliers_who_applied + suppliers_who_asked_a_question))
+
+    return invert_a_dictionary_so_supplier_id_is_key_and_brief_id_is_value(interested_suppliers)
+
+
+def invert_a_dictionary_so_supplier_id_is_key_and_brief_id_is_value(dictionary_to_invert):
+    inverted_dic = {}
+    for brief_id, list_of_supplier_ids in dictionary_to_invert.items():
+        for supplier_id in list_of_supplier_ids:
+            inverted_dic.setdefault(supplier_id, []).append(brief_id)
+
+    return inverted_dic
+
+
+def get_supplier_email_addresses_by_supplier_id(data_api_client, supplier_id):
+    response = data_api_client.find_users(supplier_id=supplier_id)
+    return [user['emailAddress'] for user in response['users']]
+
+
+def _get_link_domain(stage):
+    return env_helpers.get_api_endpoint_from_stage(stage, app='www')
+
+
+def create_context_for_supplier(stage, supplier_briefs):
+    return {
+        'briefs': [
+            {
+                'brief_title': brief['title'],
+                'brief_link': '{0}/{1}/opportunities/{2}'.format(
+                    _get_link_domain(stage),
+                    brief['frameworkFramework'], brief['id']
+                )
+            } for brief in supplier_briefs
+        ]
+    }
+
+
+def get_html_content(context):
+
+    return render_html("email_templates/suppliers_new_briefs_questions_answers.html", data={
+        "briefs": context['briefs'], "today": datetime.utcnow()
+    })
+
+
+def send_supplier_emails(email_api_key, email_addresses, supplier_context, logger):
+
+    send_email(
+        email_addresses,
+        get_html_content(supplier_context),
+        email_api_key,
+        EMAIL_SUBJECT,
+        EMAIL_FROM_ADDRESS,
+        EMAIL_FROM_NAME,
+        ["supplier-new-brief-questions-answers"],
+        logger=logger
+    )
+
+
+def main(data_api_url, data_api_token, email_api_key, stage, dry_run, exclude_supplier_ids=[]):
+    logger.info("Begin to send brief update notification emails")
+
+    # get today at 8 in the morning
```

Comment:
What time are we planning on running this script?  It feels like it's potentially going to hit the API a *lot* when it does.  I guess if the numbers of live briefs are relatively low it might not be too bad, but we should keep a close eye on if it unduly loads the API.